### ISSUE TITLE
Last line of text is squeezed together

### ISSUE THREAD

**Issue by mxx-lxg** [dompdf/dompdf#1796]
In some cases The last line of a paragraph or heading is kind of squeezed together (see screenshots). Is there a way to prevent this?

<img width="294" alt="bildschirmfoto 2018-09-13 um 11 36 02" src="https://user-images.githubusercontent.com/7644014/45480415-3be29200-b749-11e8-841e-b09d1040758a.png">

<img width="493" alt="bildschirmfoto 2018-09-13 um 11 36 30" src="https://user-images.githubusercontent.com/7644014/45480434-46049080-b749-11e8-9e54-3e58ca4765f6.png">

Version is 0.8.2

**Comment by mxx-lxg** [comment]
Update: Setting the line-height and font-size for the element in _pt_ instead of _px_ seems to fix the issue.

**Comment by bsweeney** [comment]
Interesting. Do you have some sample HTML?

**Comment by alekseykuleshov** [comment]
@bsweeney Got this issue as well, fixed by using px instead of %.

Code:
```
<?php
	require 'dompdf/vendor/autoload.php';

	use Dompdf\Dompdf;
	use Dompdf\Options;

	$options = new Options(['isHtml5ParserEnabled' => true]);

	$options->setIsRemoteEnabled(true);
	$options->setIsPhpEnabled(true);
	$dompdf = new Dompdf($options);
	$dompdf->setPaper('A4');

	$html = '
	<html>
	 <body>
		<p style="font-size:20px;line-height:80%;margin:0;">
			Aenean laoreet quis dolor non scelerisque. In eros elit, posuere at sagittis sed, rutrum et est. Praesent vel arcu enim. Morbi rhoncus libero ex, ut efficitur quam finibus eget. Duis in turpis feugiat, tempor nisl id, consequat mauris. Integer sit amet velit velit. Curabitur varius nisl sapien, sit amet porttitor metus cursus at. In nec posuere velit. Sed sit amet nulla eu mi iaculis euismod at vitae nisi. Nullam urna mauris, vestibulum non elementum ac, porta non eros. Aliquam mollis, erat eu vehicula consequat, lacus lacus gravida leo, eu accumsan augue mi nec mauris. Aenean magna mi, consectetur a neque quis, placerat porta leo. Vivamus eget lorem convallis, commodo lorem nec, malesuada tortor. Mauris ut elit risus. Cras maximus nisi ac erat consectetur, a molestie lorem laoreet. Donec gravida, purus a auctor vulputate, tortor risus fermentum urna, sit amet blandit nisl urna eget metus.
		</p>
	 </body>
	</html>';

	$dompdf->load_html($html);
	$dompdf->render();

	$dompdf->stream("some.pdf");
```

Result: [some.pdf](https://github.com/dompdf/dompdf/files/2427965/some.pdf)

**Comment by bsweeney** [comment]
Confirmed, looks like the that last line box isn't being positioned correctly. This sounds like something we've run into before, I'll see if I can find anything related.

**Comment by larsactionhero** [comment]
Same issue here. 
But: seems that not the last line is "squeezed together", looks more like that the line-height definition takes effect **only on the last row** (if more than 1 row of text). 
If there's only 1 line break in text, the line-height (here: line-height: 0.7) is displayed correctly.
By the way: If value is set it to 1, no issue appears.

![bildschirmfoto 2018-10-31 um 14 10 47](https://user-images.githubusercontent.com/11157745/47790185-ce012280-dd16-11e8-8cb7-3efea3cf5231.png)
![bildschirmfoto 2018-10-31 um 14 08 58](https://user-images.githubusercontent.com/11157745/47790186-ce99b900-dd16-11e8-9f22-c828367f292a.png)

**Comment by dami95** [comment]
I have the same problem :/
```<!DOCTYPE html>
        <html lang="de">
        <head>
            <style>
            @page {
                margin-bottom: 1cm;
            }
            
     
            body {
                font-family: Arial;
                color: #212529;
                margin: 0;
                padding: 0;
                font-size: 16px;
               
            }
            p {
                margin: 0;
                padding: 0;
            }
            
            .top {
                    
            }
            .top__left {
                
                width: 65%;
                
            }            
            .top__left__name {
                font-size: 37.5pt;
                
            }
            .top__left__name span {
            font-size: 37.5pt;
                line-height: 0.25;
            }
            
            </style>
        </head>
        <body>

        
        <div class="top">
            <div class="top__left">
                <div class="top__left__name">
                    <span>
Parmesanschnitzel zzz Tomatenrisotto und sdadsa dsa jungem Spinat
                </div>
            </div>
           
        </div>
        
             
        
</body>
</html>```

![Zrzut ekranu 2019-08-20 09 41 48](https://user-images.githubusercontent.com/724522/63327629-d4589d80-c32e-11e9-8bba-d2f17781f851.png)

---- 
After changing Tomatenrisotto -> tomatenrisotto, it looks ok:

![Zrzut ekranu 2019-08-20 09 43 22](https://user-images.githubusercontent.com/724522/63327682-f81be380-c32e-11e9-8c35-a79057bf8c1f.png)

**Comment by erjart** [comment]
I found a soluction for this problem. Place the style of element _display:inline_ and wrap element with other div, to control margin.

Ex: 
<div class="text">Lorem Ipsum Lorem Ipsum Lorem Ipsum </div>

<div class="correctwrap">
       <div class="text">Lorem Ipsum Lorem Ipsum Lorem Ipsum </div>
</div>

.text{
display:inline;
font-size:12px;
line-height:12px;
vertical-align:top;
}

.correctwrap{
margin-left:10px;     /*To create a margin from border. When you put display inline, text broke weird. */
}

**Comment by Mellthas** [comment]
This seems to be the exact same issue as #1646 and should be addressed by #2568.
